File: apps/web/lib/autonomy-control/truth.test.ts
import assert from "node:assert/strict";
import { test } from "node:test";
import { DevTruthEngine, TRUTH_SAFE_FIXTURE, TruthUnavailable } from "./truth.ts";

test("safe fixture verifies", () => {
  const r = new DevTruthEngine().verify(TRUTH_SAFE_FIXTURE);
  assert.equal(r.ok, true);
  assert.equal(r.error, undefined);
});

test("known-false claim fails without treating as unknown", () => {
  const r = new DevTruthEngine().verify("the earth is flat");
  assert.equal(r.ok, false);
  assert.equal(r.error, undefined);
});

test("empty and whitespace are unavailable, not true", () => {
  for (const input of ["", "   ", "\n\t"]) {
    const r = new DevTruthEngine().verify(input);
    assert.equal(r.ok, false);
    assert.ok(r.error instanceof TruthUnavailable);
  }
});

test("unknown claim is not verified", () => {
  const r = new DevTruthEngine().verify("unverified claim about tomorrow");
  assert.equal(r.ok, false);
  assert.ok(r.error instanceof TruthUnavailable);
});

test("deterministic repeats", () => {
  const e = new DevTruthEngine();
  assert.deepEqual(e.verify(TRUTH_SAFE_FIXTURE), e.verify(TRUTH_SAFE_FIXTURE));
  assert.deepEqual(e.verify("2+2=5"), e.verify("2+2=5"));
});

File: libs/go/pkg/autonomy/truth_test.go
package autonomy

import (
	"errors"
	"strings"
	"testing"
)

func TestTruthSafeFixture(t *testing.T) {
	ok, err := DevelopmentTruth{}.Verify(TruthSafeFixture)
	if err != nil || !ok {
		t.Fatalf("safe fixture must pass: %v %v", ok, err)
	}
}

func TestTruthKnownFalse(t *testing.T) {
	ok, err := DevelopmentTruth{}.Verify("the earth is flat")
	if err != nil || ok {
		t.Fatalf("known false must be false,nil got %v %v", ok, err)
	}
}

func TestTruthEmptyAndWhitespaceUnavailable(t *testing.T) {
	for _, in := range []string{"", "   ", "\n\t"} {
		ok, err := DevelopmentTruth{}.Verify(in)
		if ok || !errors.Is(err, ErrTruthUnavailable) {
			t.Fatalf("%q: want false+unavailable got %v %v", in, ok, err)
		}
	}
}

func TestTruthUnknownIsNotTrue(t *testing.T) {
	ok, err := DevelopmentTruth{}.Verify("unverified claim about tomorrow")
	if ok || !errors.Is(err, ErrTruthUnavailable) {
		t.Fatalf("unknown must not be true: %v %v", ok, err)
	}
}

func TestTruthDeterministic(t *testing.T) {
	a, ea := DevelopmentTruth{}.Verify(TruthSafeFixture)
	b, eb := DevelopmentTruth{}.Verify(TruthSafeFixture)
	if a != b || (ea == nil) != (eb == nil) {
		t.Fatal("not deterministic")
	}
}

func TestTruthNilEngine(t *testing.T) {
	var eng TruthEngine
	if eng != nil {
		t.Fatal("nil engine")
	}
	p := NewPlane()
	if p.Truth == nil {
		t.Fatal("NewPlane must install a development truth engine")
	}
}

func TestTruthDoesNotClaimInternetVerification(t *testing.T) {
	if !strings.Contains(DevelopmentTruth{}.Kind(), "DEVELOPMENT") {
		t.Fatal("must disclose development engine")
	}
}
